BlueprintUI/Sources/Layout/Builder/StackChild.swift

```diff
@@ -0,0 +1,91 @@
+import UIKit
+
+/// `StackChild` is a wrapper which allows an element to define its StackLayout traits and Keys.
+/// This struct is particularly useful when working with `@ElementBuilder`.
+/// By default, elements will default to a nil key and the default `StackLayout.Traits` initializer.
+/// `@StackElementBuilder` will check every child to see if it can be type cast to a `StackChild`
+/// and then pull of the given traits and key and then apply those to the stack
+public struct StackChild {
+    public let element: Element
+    public var traits: StackLayout.Traits
+    public var key: AnyHashable?
+    
+    public struct Sizing {
+        public static let fixed: Self = .init(growPriority: 0, shrinkPriority: 0)
+        public static let flexible: Self = .init(growPriority: 1, shrinkPriority: 1)
+
+        public var growPriority: CGFloat
+        public var shrinkPriority: CGFloat
+        
+        public init(growPriority: CGFloat = 1, shrinkPriority: CGFloat = 1) {
```

Comment:
I would leave the defaults off here. This API is a super big footgun (most people don't understand it at all), so I'd make them pass this explicitly.

Going further; since this is an API split; I wonder if we should take this opportunity to make the sizing for result builders only be able to take in Bools for these.

Reply:
Hm granted I'm not as familiar with the API, but don't these priorities allow you to balance priorities among multiple elements similar to `contentHugging` and `compressionResistence` in UIKit? Wouldn't a bool only allow comparison between two elements? Unless you mean to change it to `allowShrinking: Bool` and `allowGrowing: Bool`.

Reply:
> don't these priorities allow you to balance priorities among multiple elements similar to `contentHugging` and `compressionResistence` in UIKit? 

Kind of, but in practice, the priority algorithm (documented on `StackElement.add`) doesn't do what you probably expect, and so we often recommend sticking to either 0/0 for "fixed" items and 1/1 for "flexible" items. The `addFixed` and `addFlexible` methods on stacks were added to encourage that.

> Wouldn't a bool only allow comparison between two elements? Unless you mean to change it to `allowShrinking: Bool` and `allowGrowing: Bool`.

I think what Kyle is suggesting is to limit this API to 2 discrete cases (fixed & flexible) rather than exposing the numbers at all. And I agree! The underlying values are still stored on the traits if we want to expose them later.